basket/news/tasks.py

```diff
@@ -256,11 +256,17 @@ def fxa_email_changed(data):
         # message older than our last update for this UID
         return
 
+    cache.set(cache_key, ts, 7200)  # 2 hr
+
     sfmc.upsert_row('FXA_EmailUpdated', {
         'FXA_ID': fxa_id,
         'NewEmailAddress': email,
     })
-    cache.set(cache_key, ts, 7200)  # 2 hr
+
+    # Update SFDC
+    user_data = get_user_data(fxa_id=fxa_id, extra_fields=['id'])
```

Comment:
And I think we should do the SFDC update first because the SFMC one just inserts rows, so if this task is retried due to an SFDC issue there will be dupes in SFMC. Not a big deal though.